src/gc.c

```diff
@@ -137,8 +142,10 @@ void GCContext_Stop(GCContext* gc) {
     // free gc
     destroyCallback(gc);
   } else {
-    // GC is running, we add a task to the thread pool to free it
-    redisearch_thpool_add_work(gcThreadpool_g, destroyCallback, gc, THPOOL_PRIORITY_HIGH);
+    // GC is running, or is about to run.
```

Comment:
Let's elaborate on what it means that "GC is running, or is about to run". From looking at `Redis` code, we first execute the timer callback, and then we remove it from the timers' collection. Meaning, if `RM_StopTimer` didn't return `REDISMODULE_OK`, it means that the timer was removed from the collection. However, the reason why we cannot assume that GC ran is because we are executing it from within another thread. Hence, the `timerCallback` might have ended (and timer removed from the collection as a result) while `threadCallback` is running (or about to run) - and this is why we cannot call `destoryCallback` here. Am I right?

Reply:
I think this is not true, as I mentioned to you privately. At this point, "about to run" means that the periodic job is already in the Gcs queue

Reply:
Why not true? I'm not contradicting what you said (I think). Anyway, let's discuss it offline - no need to block this PR as we agree on the solution.